lib/shoppingcart.rb

```diff
@@ -0,0 +1,59 @@
+require './lib/shoppingcart'
+
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @details = nil
```

Comment:
no need to set this in here since you have a method called details that is creating and returning the desired hash. If your code ever tried accessing `@details` before ever calling the `details` method, then it would get a nil value, so for that reason, I'd stick to just having the details method.